pkg/clients/iam/iamrole.go

```diff
@@ -137,21 +137,37 @@ func CreatePatch(in *iam.Role, target *v1beta1.IAMRoleParameters) (*v1beta1.IAMR
 }
 
 // IsRoleUpToDate checks whether there is a change in any of the modifiable fields in role.
-func IsRoleUpToDate(in v1beta1.IAMRoleParameters, observed iam.Role) (bool, error) {
+func IsRoleUpToDate(in v1beta1.IAMRoleParameters, observed iam.Role) (bool, string, error) {
 	generated, err := copystructure.Copy(&observed)
 	if err != nil {
-		return true, errors.Wrap(err, errCheckUpToDate)
+		return true, "", errors.Wrap(err, errCheckUpToDate)
 	}
 	desired, ok := generated.(*iam.Role)
 	if !ok {
-		return true, errors.New(errCheckUpToDate)
+		return true, "", errors.New(errCheckUpToDate)
 	}
 
 	if err = GenerateIAMRole(in, desired); err != nil {
-		return false, err
+		return false, "", err
 	}
 
-	return cmp.Equal(desired, &observed, cmpopts.IgnoreInterfaces(struct{ resource.AttributeReferencer }{})), nil
+	isRoleUpToDate := cmp.Equal(desired, &observed, cmpopts.IgnoreInterfaces(struct{ resource.AttributeReferencer }{}))
```

Comment:
Would using `cmp.Diff` once and checking whether its output is empty string or not to infer equality work?

Reply:
Yes! I just checked and cmp.Diff == "" can be used instead